vcs/internal/script.go

```diff
@@ -0,0 +1,53 @@
+package internal
+
+import (
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"time"
+)
+
+var gen rand.Source
+
+func init() {
+	gen = rand.NewSource(time.Now().UnixNano())
+}
+
+// Constructs platform-specific temporary script file with a given prefix
+// On Windows such a file must have .bat extension
+func ScriptFile(prefix string) (string, error) {
+
+	if runtime.GOOS == "windows" {
+		for {
+			tempFile := filepath.Join(os.TempDir(), prefix+strconv.FormatInt(gen.Int63(), 36)+".bat")
```

Comment:
I think he's doing this because `ioutil.TempFile()` doesn't let you specify the extension, but .bat is needed.

I highly suggest an alternative of using `ioutil.TempDir()` and creating a file with the desired extension inside. See [here](https://github.com/shurcooL/goe/blob/3bcc0e08dda8182005ab103f77e7bf8749d1cf7e/goe.go#L33-L50) for an example.